python/desc/sims/GCRCatSimInterface/CatalogClasses.py

```diff
@@ -4,15 +4,91 @@
 from lsst.utils import getPackageDir
 from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogSersic2D
 from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogZPoint
+from lsst.sims.catUtils.exampleCatalogDefinitions import PhoSimCatalogSN
 from lsst.sims.catUtils.mixins import VariabilityAGN
 from lsst.sims.catalogs.decorators import cached, compound
 from lsst.sims.catUtils.mixins import EBVmixin
 
-__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN"]
+from lsst.sims.catalogs.db import fileDBObject
+
+__all__ = ["PhoSimDESCQA", "PhoSimDESCQA_AGN", "SNFileDBObject",
+           "DC2PhosimCatalogSN"]
 
 #########################################################################
 # define a class to write the PhoSim catalog; defining necessary defaults
 
+class SNFileDBObject(fileDBObject):
+    """
+    Use FileDBObject to provide CatalogDBObject functionality for SN
+    with host galaxies from protoDC2 output to csv files before
+    """
+    dbDefaultValues = {'varsimobjid':-1,
+                       'runid':-1,
+                       'ismultiple':-1,
+                       'run':-1,
+                       'runobjid':-1}
+
+    # These types should be matched to the database.
+    #: Default map is float.  If the column mapping is the same as the
+    # column name, None can be specified
+
+    columns = [('raJ2000', 'snra_in*PI()/180.'),
+               ('decJ2000', 'sndec_in*PI()/180.'),
+               ('Tt0', 't0_in'),
+               ('Tx0', 'x0_in'),
+               ('Tx1', 'x1_in'),
+               ('Tc', 'c_in'),
+               ('id', 'snid_in'),
+               ('Tredshift', 'z_in'),
+               ('redshift', 'z_in'),
+               ('Tgaltileid', 'galtileid')
+              ]
+class DC2PhosimCatalogSN(PhoSimCatalogSN):
```

Comment:
There should be space between class definitions (but this problem will, of course, go away if you move `SNFileDBObject` to `DatabaseEmulator.py`)